pkg/processor/metricfrequencyprocessor/strategy.go

```diff
@@ -0,0 +1,182 @@
+package metricfrequencyprocessor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"math"
+	"sort"
+	"time"
+)
+
+const (
+	float64EqualityThreshold = 1e-9
+	minPointAccumulationTime = time.Minute * 15
+	category1ReportFrequency = time.Second * 30
+	category2ReportFrequency = time.Minute * 2
+	category3ReportFrequency = time.Minute * 5
+	safetyInterval           = time.Second * 1
+
+	iqrAnomalyCoef            = 1.5
+	variationIqrThresholdCoef = 4
+)
+
+type MetricSieve struct {
+	metricCache  MetricCache
+	lastReported map[string]pdata.Timestamp
+}
+
+func (fs *MetricSieve) Sift(metric pdata.Metric) bool {
+	if metric.DataType() == pdata.MetricDataTypeDoubleGauge {
+		return fs.siftDropGauge(metric)
+	} else {
+		return false
+	}
+}
+
+func (fs *MetricSieve) siftDropGauge(metric pdata.Metric) bool {
+	metric.DoubleGauge().DataPoints().RemoveIf(fs.siftDataPoint(metric.Name()))
+
+	return metric.DoubleGauge().DataPoints().Len() == 0
+}
+
+func (fs *MetricSieve) siftDataPoint(name string) func(pdata.DoubleDataPoint) bool {
+	return func(dataPoint pdata.DoubleDataPoint) bool {
+		cachedPoints := fs.metricCache.List(name)
+		fs.metricCache.Register(name, dataPoint)
+		lastReported, _ := fs.lastReported[name]
```

Comment:
Don't use the shortcuts :) check the second value

Reply:
it's redundant, but I made additional check here